kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Utils.java

```diff
@@ -345,32 +346,8 @@ public static String toUrlEncoded(String str) {
     return null;
   }
 
-  /**
-   *  todo: we should unify this with {@link Pluralize}
-   */
   public static String getPluralFromKind(String kind) {
-    StringBuilder pluralBuffer = new StringBuilder(kind.toLowerCase(Locale.ROOT));
-    switch (kind) {
-      case "ComponentStatus":
-      case "Ingress":
-      case "RuntimeClass":
-      case "PriorityClass":
-      case "StorageClass":
-        pluralBuffer.append("es");
-        break;
-      case "NetworkPolicy":
-      case "PodSecurityPolicy":
-      case "ServiceEntry": // an Istio resource. Needed as getPluralFromKind is currently not configurable #2489
-        // Delete last character
-        pluralBuffer.deleteCharAt(pluralBuffer.length() - 1);
-        pluralBuffer.append("ies");
-        break;
-      case "Endpoints":
-        break;
-      default:
-        pluralBuffer.append("s");
```

Comment:
Related to the new test failure,

should we preserver the code block and set default to:
```java
    default:
        return Pluralize.toPlural(kind.toLowerCase(Locale.ROOT));
```

@rohanKanojia @metacosm ???